src/wrapper/simulation_object_updating_component.py

```diff
@@ -11,8 +14,40 @@ class SimulationObjectUpdatingComponent(Component):
 
     _simulation_objects = None
 
-    def __init__(self, *args, **kwargs):
-        Component.__init__(self, *args, **kwargs)
+    @property
+    def simulation_objects(self) -> List["SimulationObject"]:
+        """Returns a list of all objects in the simulation
+
+        :return: the objects in the simulation
+        """
+        return self._simulation_objects
+
+    @property
+    def signals(self) -> List["Signal"]:
+        """Returns all signals in the simulation
+
+        :return: The signals in the simulation
+        """
+        return [
+            x
+            for x in self._simulation_objects
+            if isinstance(x, src.wrapper.simulation_objects.Signal)
+        ]
+
+    @property
+    def tracks(self) -> List["Track"]:
+        """Returns all tracks in the simulation
+
+        :return: The tracks in the simulation
+        """
+        return [
+            x
+            for x in self._simulation_objects
+            if isinstance(x, src.wrapper.simulation_objects.Track)
+        ]
+
+    def __init__(self, logger=None):
+        Component.__init__(self, priority=10, logger=logger)
```

Comment:
Probably it doesn't make a difference here but the cleaner solution would be to call `super().__init__`. 
See this Post: https://stackoverflow.com/questions/42413670/whats-the-difference-between-super-and-parent-class-name

super calls in python are tricky 😬